torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -3912,13 +3910,13 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::barrier(const BarrierOptions& opts) {
     // ensure that each process is on a different GPU
     auto numGPUs = at::cuda::getNumGPUs();
     int16_t deviceIdx = static_cast<int16_t>(rank_ % numGPUs);
-    LOG(INFO)
+    LOG(WARNING)
         << logPrefix()
         << c10::str(
-               " using GPU ",
+               "using GPU ",
                deviceIdx,
                " to perform barrier as devices used by this process are currently unknown. ",
-               "This can potentially cause a hang if this rank to GPU mapping is incorrect.",
+               "This can potentially cause a hang if this rank to GPU mapping is incorrect. ",
```

Comment:
`logPrefix()` has a postfix space so the following space is not necessary. The space at this line is to seperate current sentence from the following one.

Before:
```
[rank1]:[I719 15:04:29.184536462 ProcessGroupNCCL.cpp:3913] [PG 0 Rank 1]  using GPU 1 to perform barrier as devices used by this process are currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect.Specify device_ids in barrier() to force use of a particular device.
[rank0]:[I719 15:04:29.184536336 ProcessGroupNCCL.cpp:3913] [PG 0 Rank 0]  using GPU 0 to perform barrier as devices used by this process are currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect.Specify device_ids in barrier() to force use of a particular device.
```
After:
```
[rank1]:[W719 14:59:17.550906449 ProcessGroupNCCL.cpp:3913] [PG 0 Rank 1] using GPU 1 to perform barrier as devices used by this process are currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect. Specify device_ids in barrier() to force use of a particular device.
[rank0]:[W719 14:59:17.555036961 ProcessGroupNCCL.cpp:3913] [PG 0 Rank 0] using GPU 0 to perform barrier as devices used by this process are currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect. Specify device_ids in barrier() to force use of a particular device.
```